deploy/PSM3Init.sol

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+import { PSM3 } from "src/PSM3.sol";
+
+import { IERC20 } from "erc20-helpers/interfaces/IERC20.sol";
+
+library PSM3Init {
+
+    function init(address psm, address asset0) external {
+        IERC20(asset0).approve(psm, 1e18);
+        PSM3(psm).deposit(asset0, address(0), 1e18);
+    }
```

Comment:
This should not be done as part of the init phase. It should be done right after deployment. In fact there is no init required because the PSM3 is permissionless and doesn't require spell setup. A deploy script is sufficient.